### ISSUE TITLE
Many tests fail due to MPI not being initialized

### ISSUE THREAD

**Issue by jbigot** [pdidev/pdi#93]
The logging infrastructure now depends on MPI (call to `MPI_Comm_rank`) so unit tests that do not initialize MPI now fail with some versions of OpenMPI (`DataDescTest` & `ContextTest`)

```
*** The MPI_Comm_rank() function was called before MPI_INIT was invoked.
*** This is disallowed by the MPI standard.
*** Your MPI job will now abort.
```

**Comment by jbigot** [comment]
mentioned in commit eb965686b7cdc7fe94b88a9312bb42a0a42e39e2

**Comment by jbigot** [comment]
mentioned in commit 73ad72f61f21317cd3db7d72191d1a96ec7e4065

**Comment by jbigot** [comment]
closed via merge request !143

**Comment by jbigot** [comment]
closed via commit 73ad72f61f21317cd3db7d72191d1a96ec7e4065

**Comment by jbigot** [comment]
closed via merge request !142
